storybook/stories/playground/box/index.js

```diff
@@ -32,12 +32,10 @@ export default function EditorBox() {
 				value={ blocks }
 				onInput={ updateBlocks }
 				onChange={ updateBlocks }
-				settings={ {
-					hasFixedToolbar: true,
-				} }
 			>
-				<BlockTools />
-				<BlockCanvas height="100%" styles={ editorStyles } />
+				<BlockTools>
```

Comment:
Why block tools need to wrap BlockCanvas? why shouldn't we just embed it in BlockCanvas? (Goal is to have the least possible code to write for developers).

Also I'm assuming that this change is going to make the toolbar "contextual" (by the block)? right? how can a consumer render the toolbar in its desired position (like a fixed toolbar in the post editor) or elsewhere?

Reply:
BTw, the "box" playground is meant to create something like "TinyMCE" where the toolbar is fixed at the top (something like the isolated block editor experiment). So this change is actually undoing that.

Reply:
Ok It seems the linked PR is what broke these stories. I commented on it.  https://github.com/WordPress/gutenberg/pull/55787#issuecomment-1827596371

Reply:
Are you sure it was working before that PR?

Reply:
@ellatrix yes, I did create these two stories myself when I was writing the platform-docs to confirm how third-party developers should do it.

Reply:
Strange! In any case `BlockTools` is not meant to be used like that. We can build it into `BlockCanvas` I guess. It also adds shortcuts, the inserter/appender, etc.

Reply:
Yeah we need three things:

 - By default all the shortcuts, appender need to work without any additional code than `BlockCanvas`
 - By default the block editor should show contextual toolbars 
 - We should be able to disable contextual toolbars (setting in the block editor provider) and render a block toolbar using `BlockToolbar` component (or something else)

Reply:
Yep, agreed. Also I tried using the top toolbar setting, and BlockTools still renders a toolbar if the screen size is not "large".

Reply:
> I tried using the top toolbar setting, and BlockTools still renders a toolbar if the screen size is not "large".

This was intentional to match the existing behavior. The block toolbar is supposed to sit underneath the header at large screens for either popover or top toolbar settings.

Reply:
> This was intentional to match the existing behavior. The block toolbar is supposed to sit underneath the header at large screens for either popover or top toolbar settings.

In some third-party editors, there might not be any header. So when we touch these APIs, we should consider these use-cases. Is this behavior something that should be baked in or something that's left to the consumer. 

In this case, there's no "Header" component exposed by block-editor, so probably not something we can bake it in a generic way.

Reply:
What are the available presentation options for `<BlockToolbar />` in `<BlockTools />`? My understanding was popover until smaller screen, which gets fixed to the top of the editor pane. Instead, should it be `<BlockTools blockToolbar="popover|fixed|none" />` and we pass the option we want into `<BlockTools />`?

Then, for our editor, the flow looks like:

```
// begin pseudo code
const isTopToolbar = getSettings().hasFixedToolbar;
const isLargeViewport = useViewportMatch( 'medium' );

let blockToolbarDisplay = 'popover';

if ( isLargeViewport ) {
  // pinned to top of screen
  blockToolbarDisplay = "fixed";
} else if ( isTopToolbar ) {
  // Handled by the external header. BlockToolbar is never rendered within `<BlockTools />`
  blockToolbarDisplay = "none";
}

<BlockTools blockToolbar={ blockToolbarDisplay } />
```

Reply:
Let's not use `BlockTools` please, it's also needed for shortcuts and inline inserter. `BlockTools` should be absorbed into `BlockCanvas` and we should export another component to create a top toolbar.

Reply:
Oh yeah, no intention of exporting `<BlockTools />` for the top toolbar. My code example was to remove the concept of a top toolbar from `<BlockTools />` due to @youknowriad's explanation: 

> If hasFixedToolbar is false (default), the BlockContextualToolbar is rendered, otherwise it's not and you are (consumer of block-editor package, in our case edit-* packages) responsible of rendering the block toolbar by using the <BlockToolbar /> component directly.
> https://github.com/WordPress/gutenberg/pull/55787#discussion_r1387775559

`<BlockToolbar />` is the component we're trying to get to be easily useful for the top toolbar. Right now we had to privately export `<ContextualBlockTools />`, so https://github.com/WordPress/gutenberg/pull/56335 is trying to make `<BlockToolbar />` useful on its own instead of relying on `<ContextualBlockTools />`.